src/crimson/common/interruptible_future.h

```diff
@@ -1170,7 +1164,7 @@ struct interruptor
   [[gnu::always_inline]]
   static auto wrap_function(Func&& func) {
     return [func=std::forward<Func>(func),
-	    interrupt_condition=interrupt_cond<InterruptCond>]() mutable {
+	    interrupt_condition=interrupt_cond<InterruptCond>.interrupt_cond]() mutable {
```

Comment:
Why there is no ceph_assert(interrupt_cond<InterruptCond>.interrupt_cond);?

If it's possible here that interrupt_cond can be null, then which scenario is that?

Ditto for the below places where `interrupt_cond` is captured but not asserted with existance.